example/airtable/events/register.php

```diff
@@ -3,22 +3,39 @@
 namespace RemoteDataBlocks\Example\Airtable\Events;
 
 use RemoteDataBlocks\Integrations\Airtable\AirtableDataSource;
+use RemoteDataBlocks\Integrations\Airtable\AirtableIntegration;
 
-require_once __DIR__ . '/inc/queries/class-airtable-get-event-query.php';
-require_once __DIR__ . '/inc/queries/class-airtable-list-events-query.php';
-
-function register_airtable_events_block() {
-	$block_name = 'Conference Event';
+function register_airtable_events_block(): void {
 	$access_token = \RemoteDataBlocks\Example\get_access_token( 'airtable_events' );
 
-	$airtable_data_source = AirtableDataSource::create( $access_token, 'appVQ2PAl95wQSo9S', [], 'Conference Events' );
-	$airtable_get_event_query = new AirtableGetEventQuery( $airtable_data_source );
-	$airtable_list_events_query = new AirtableListEventsQuery( $airtable_data_source );
+	if ( empty( $access_token ) ) {
+		return;
+	}
+
+	$airtable_data_source = AirtableDataSource::from_array( [
+		'service_config' => [
+			'__version' => 1,
+			'access_token' => $access_token,
+			'base' => [
+				'id' => 'appVQ2PAl95wQSo9S',
+				'name' => 'Conference Events',
+			],
+			'display_name' => 'Conference Events',
+			'tables' => [], // AirtableDataSource does not formally provide queries.
+		],
+	] );
+
+	$block_options = [
+		'pages' => [
```

Comment:
Is it necessary for page registration to be part of the block registration? I mainly ask because we might want to have overrides for multiple blocks on a page.

Reply:
Confirmed you can do this now. The page is only created once if you use the same page slug for multiple blocks. 

We really need some deeper product exploration around pages, though, because I don't really know what use cases we are solving.